### ISSUE TITLE
Search Dependancy

### ISSUE THREAD

**Issue by blainsmith** [oddnetworks/oddworks#20]
So a feature of Oddworks is "search". Right now on version 1 we solely rely on Elasticsearch and tightly integrated into our current catalog. With version 2 on the horizon we wanted to offer multiple options for indexing and searching content within the system.

Right now we have a concept of a **store** that get, set, delete, and list resources based on type.

So my question to you all is now that we have a **store** that saves collections, videos, etc. that the catalog uses to do its work. Do we also treat the search db as a separate **store** or is it a direct dependancy of the catalog that essentially enables search.

If it is the former, then does the search **store** just extend existing API with 2 new pattern? **{role: 'store', cmd: 'index'}** and **{role: 'store', cmd: 'search'}**

```js
redisStore.initialize(bus, {redis, types: ['collection', 'promotion', 'video', 'view']});
elasticsearchStore.initialize(bus, {elasticsearch, types: ['collection', 'video']});
```

If you like the latter, then a configurable option of the catalog would be a search URI which points to one of the many fulltext search dbs out there like Elasticsearch, Solr, etc.

```js
catalogService.initialize(bus, {searchUri: 'elasticsearch://localhost:9200'}),
```

Behind the scenes the Catalog Service will load the correct client based on the protocol used (elasticsearch://, solr://) or some configurable identifier.

Our goal is to make this as easy as possible so any insight anyone has is welcome!

**Comment by clarkefoley** [comment]
Just jumping in to say this discussion is relevant to my interests per https://github.com/oddnetworks/aws-elasticsearch-transporter/issues/2 and https://github.com/oddnetworks/aws-elasticsearch-transporter/issues/1

Maybe a possible solution is adding a organization / device column that when a search query from a specific device hits the store, it only returns results for that org and device combo. I'll butt out now :)

**Comment by blainsmith** [comment]
This has been implemented in 2.0.0.
